feat : update api nodes to use jwtToken innerAPI

YunYeongChang opened the pull request:
@Malang2  new JWT Verification Process

Closes #9

Commit:
feat : change user schema and apply to nodes

Commit:
feat : add jwttoken manage functions to jwtauth folder

+ routes/jwtauth/jwtBlock.js -> block used,expired jwttoken
+ routes/jwtauth/jwtRefresh.js -> refresh accesstoken with refreshtoken
+ routes/jwtauth/jwtSign.js -> export jwtsign function
+ routes/jwtauth/jwtVerify.js -> add verification with user status
-> routes/jwtauth/jwtdecode.js -> move to jwtauth folder

Commit:
feat : update api nodes to use new jwttoken inner api

* routes/auth/login.js
* routes/auth/signup.js
* routes/auth/social-kakao.js -> use new inner api
> routes/auth/jwtToken.js
* swagger.yaml -> update output model

YunYeongChang commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- routes/jwtauth/jwtgetUser.js  3
         

Clones added
============
- routes/post/new_article.js  1
- routes/post/mod_article.js  1
- routes/jwtauth/jwtRefresh.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

YunYeongChang commented on routes/jwtauth/jwtgetUser.js:
```diff
@@ -0,0 +1,54 @@
+    };
+
+    //#HANDLE THIS NODE MORE WHEN OPTIONS ARE EXIST
+    if (!(objectID === undefined || objectID === null)) {
```
Codacy found an issue: ['objectID' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- routes/jwtauth/jwtgetUser.js  3
         

Clones added
============
- routes/post/mod_article.js  1
- routes/jwtauth/jwtRefresh.js  1
- routes/post/new_article.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

rycont reviewed (dismissed):
LGTM✌

rycont commented on models/jwtblock.js:
```diff
@@ -0,0 +1,10 @@
+import { model, Schema } from "mongoose";
+
+const jwtBlock = new Schema({
+  "blockdate": String,
```
Doublequotes not required!

YunYeongChang replied:
@rycont Thank you for helpful information

Commit:
feat : implement jwt_community authentication function

+ models/group/* -> new model for community, authentication
+ models/user.js -> link privilege object id to user object
+ routes/jwtauth/jwtVerifyAccess.js -> check user accessibility
* routes/post/* -> apply jwtcommunity auth options to node
+ routes/jwtauth/jwtgetUser.js -> export validation code
* swagger.yaml -> update swagger document

Malang2 commented:
great

Commit:
feat : refresh accessToken using refreshToken in post-node

YunYeongChang commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- routes/post/load_comments.js  7
- routes/post/new_article.js  11
- routes/post/mod_article.js  12
- routes/post/load_articles.js  7
- routes/jwtauth/jwtVerifyAccess.js  17
- routes/post/load_editlogs.js  10
- routes/jwtauth/jwtgetUser.js  9
- routes/post/mod_comment.js  9
- routes/post/new_comment.js  8
- routes/jwtauth/jwtVerify.js  7
         

Clones added
============
- routes/post/load_comments.js  1
- routes/post/new_article.js  1
- routes/post/mod_article.js  2
- routes/post/load_articles.js  1
- routes/jwtauth/jwtRefresh.js  1
- routes/post/new_comment.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

YunYeongChang commented on routes/jwtauth/jwtRefresh.js:
```diff
@@ -0,0 +1,68 @@
+import authLog from "../../models/authlog";
+import User from "../../models/user";
+  
+const jwtRefresh = async (req, accesstoken, refreshtoken) => {
```
Codacy found an issue: [Async arrow function has a complexity of 8. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+    return _response;
+};
+
+const jwtVerifyAccess = async (userid, type, target) => {
```
Codacy found an issue: [Async arrow function has a complexity of 17. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
Codacy found an issue: [Async arrow function has a complexity of 17. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        if (!_community) return returnResult(null, _community);
+
+        //#FIND USER PRIVILEGE IN COMMUNITY MEMBER LIST
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
```
Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
```
Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerify.js:
```diff
@@ -0,0 +1,29 @@
+import jwtBlock from "../../models/jwtblock";
+import User from "../../models/user";
+  
+const jwtVerify = async (token, ignoreblock) => {
```
Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
+        if (!privilege) return returnResult(null, _privilege);
+        return returnResult(privilege.permission, null);
```
Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
+        if (!privilege) return returnResult(null, _privilege);
```
Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtgetUser.js:
```diff
@@ -0,0 +1,53 @@
+import jwtVerifyAccess from "./jwtVerifyAccess";
+import moment from "moment";
+
+const jwtgetUser = async (req, jwtToken, options) => {
```
Codacy found an issue: [Async arrow function has a complexity of 9. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
Codacy found an issue: [Async arrow function has a complexity of 9. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 9
           

Complexity increasing per file
==============================
- routes/post/load_comments.js  9
- routes/post/load_editlogs.js  12
- routes/jwtauth/jwtgetUser.js  9
- routes/post/mod_comment.js  11
- routes/post/new_article.js  13
- routes/jwtauth/jwtRefresh.js  11
- routes/post/mod_article.js  14
- routes/jwtauth/jwtVerify.js  7
- routes/jwtauth/jwtVerifyAccess.js  17
- routes/post/load_articles.js  9
- routes/post/new_comment.js  10
         

Clones added
============
- routes/post/load_comments.js  3
- routes/post/load_editlogs.js  2
- routes/post/mod_comment.js  1
- routes/post/new_article.js  2
- routes/jwtauth/jwtRefresh.js  1
- routes/post/mod_article.js  3
- routes/post/load_articles.js  3
- routes/post/new_comment.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

YunYeongChang commented on routes/jwtauth/jwtRefresh.js:
```diff
@@ -0,0 +1,71 @@
+import authLog from "../../models/authlog";
+import User from "../../models/user";
+  
+const jwtRefresh = async (req, accesstoken, refreshtoken) => {
```
Codacy found an issue: [Async arrow function has a complexity of 8. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
```
Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
solved

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        if (!_community) return returnResult(null, _community);
+
+        //#FIND USER PRIVILEGE IN COMMUNITY MEMBER LIST
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
```
Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
solved

YunYeongChang commented on routes/post/load_comments.js:
```diff
@@ -1,23 +1,32 @@
 
 const router = Router();
-router.get ("/:target", async (req,res) => {//#CHECK DATABASE STATE AND WHETHER PROVIDED POST DATA IS VALID 
+router.get ("/:target", async (req,res) => {
```
Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang commented on routes/jwtauth/jwtVerifyAccess.js:
```diff
@@ -0,0 +1,43 @@
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
+        if (!privilege) return returnResult(null, _privilege);
```
Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

YunYeongChang replied:
solved

Commit:
Merge branch 'master' into develop

MinSeungHyun reviewed (approved):
👍

alsh02 reviewed (approved):
so beautiful code
